wormhole-connect/src/utils/sdkv2.ts

```diff
@@ -50,3 +94,149 @@ export async function getDecimals(
   const wh = await getWormholeContextV2();
   return await wh.getDecimals(chain, token.address);
 }
+
+type ReceiptWithAttestation<AT> =
+  | AttestedTransferReceipt<AT>
+  | RedeemedTransferReceipt<AT>
+  | DestinationQueuedTransferReceipt<AT>
+  | CompletedTransferReceipt<AT>;
+
+// `parseReceipt` is used when we resume a transaction to get the transaction details
+// from the VAA. Each protocol has different data in its VAAs and this parses them
+// into the common internal format used by Connect: `TransferInfo`
+export function parseReceipt(
+  route: Route,
+  receipt: ReceiptWithAttestation<any>,
+): TransferInfo | null {
+  switch (route) {
+    case Route.Bridge:
+      return parseTokenBridgeReceipt(
+        receipt as ReceiptWithAttestation<TokenBridge.TransferVAA>,
+      );
+    case Route.CCTPManual:
+      return parseCCTPReceipt(
+        receipt as ReceiptWithAttestation<CircleTransfer.CircleAttestationReceipt>,
+      );
+    default:
+      throw new Error(`Unknown route type ${route}`);
+  }
+}
+
+const parseTokenBridgeReceipt = (
+  receipt: ReceiptWithAttestation<TokenBridge.TransferVAA>,
+): TransferInfo => {
+  const txData: Partial<TransferInfo> = {
+    toChain: config.sdkConverter.toChainNameV1(receipt.to),
+    fromChain: config.sdkConverter.toChainNameV1(receipt.from),
+  };
+
+  if ('originTxs' in receipt && receipt.originTxs.length > 0) {
+    txData.sendTx = receipt.originTxs[receipt.originTxs.length - 1].txid;
+  } else {
+    throw new Error("Can't find txid in receipt");
+  }
+
+  /* @ts-ignore */
+  // TODO typescript is complaining about the second attestation property not existing when it does
+  const { payload } = receipt.attestation.attestation;
+
+  if (!payload.token) {
+    throw new Error(`Attestation is missing token.`);
+  }
+
+  if (payload.token) {
+    const tokenIdV2 = Wormhole.tokenId(
+      payload.token.chain,
+      payload.token.address,
+    );
+    const tokenV1 = config.sdkConverter.findTokenConfigV1(
+      tokenIdV2,
+      config.tokensArr,
+    );
+
+    if (!tokenV1) {
+      // This is a token Connect is not aware of
+      throw new Error('Unknown token');
+    }
+
+    const fromChain = config.sdkConverter.toChainNameV1(receipt.from);
+    const fromChainConfig = config.chains[fromChain];
+    const decimals =
+      tokenV1!.decimals[fromChainConfig!.context] || tokenV1!.decimals.default;
+
+    txData.tokenDecimals = decimals;
+    txData.amount = amount.display({
```

Comment:
I resumed this tx from avalanche https://avascan.info/blockchain/c/tx/0x9f5905c58e2b234e680649e232fabfa9b5345d0a65c2c708f7075de0e7961f91 and the UI is showing an amount of 1e-12 instead of 0.01. I also saw this error in the console:
```
chunk-QU4HNYML.js?v=9dd551e1:2618 Uncaught (in promise) Error: chunk is shorter than expected: 1048 < 1060
    at updateOffset (chunk-QU4HNYML.js?v=9dd551e1:2618:11)
    at deserializeLayoutItem (chunk-QU4HNYML.js?v=9dd551e1:2703:7)
    at internalDeserializeLayout (chunk-QU4HNYML.js?v=9dd551e1:2627:47)
    at deserializeLayout (chunk-QU4HNYML.js?v=9dd551e1:2610:19)
    at chunk-PPHDD64C.js?v=9dd551e1:2622:14
    at deserializePayload (chunk-PPHDD64C.js?v=9dd551e1:2627:5)
    at deserialize (chunk-PPHDD64C.js?v=9dd551e1:2601:5)
    at getVaa (chunk-X4S4DBQ4.js?v=9dd551e1:2454:10)
```
I also don't see the button to redeem after connecting my wallet

Reply:
![image](https://github.com/user-attachments/assets/0ef689c9-c993-43c7-aa09-cdbbe53e0f06)